observer/circularQueueNodesProvider.go

```diff
@@ -1,18 +1,15 @@
 package observer
 
 import (
-	"sync"
-
 	"github.com/multiversx/mx-chain-proxy-go/data"
+	"github.com/multiversx/mx-chain-proxy-go/observer/mapCounters"
 )
 
 // circularQueueNodesProvider will handle the providing of observers in a circular queue way, guaranteeing the
 // balancing of them
 type circularQueueNodesProvider struct {
 	*baseNodeProvider
-	countersMap        map[uint32]uint32
-	counterForAllNodes uint32
-	mutCounters        sync.RWMutex
+	positionsHolder *mapCounters.MapCountersHolder
```

Comment:
interface instead?

Reply:
done. but didn't pass it to the constructor since the functionality of the `circularQueueNodesProvider` depends a lot on the actual implementation, so stubs in tests wouldn't help.